### PR TITLE
resolved issues #16

### PR DESCRIPTION
**Pull request by sunju0123**
- update page table
    url column :  add  constraints "unique"
- add try catch in insertUrl method

resolve check duplication urls. 
Now automatically check url when using insertUrl Method

and ..
develop getUnreadUrlList method
update page table, read column : add default value 0 (means unread)
change method name :  getUrlList -> getAllUrlList

### CODE REVIEW

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java [hunk 1]
```diff
@@ -76,8 +76,12 @@ public void setWordCount(String url, int wordCount) {
...
   public void insertScriptedData(String url) {
     SQLiteDatabase dataBase = getWritableDatabase();
-    dataBase.execSQL("INSERT INTO PAGE (URL) VALUES ('" + url + "');");
-    dataBase.close();
+    try {
+      dataBase.execSQL("INSERT INTO PAGE (URL) VALUES ('" + url + "');");
+      dataBase.close();
+    } catch (Exception exception) {
+      dataBase.close();
+    }
```
Comment (xhae) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: `try {
dataBase.execSQL("...");
} catch {
}
dataBase.close();`

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java [hunk 2]
```diff
@@ -47,14 +47,27 @@ public int getReadPageCount() {
...
-  public ArrayList<ScriptedURL> getUrlList() {
+  public ArrayList<ScriptedURL> getAllUrlList() {
     SQLiteDatabase dataBase = getReadableDatabase();
     ArrayList<ScriptedURL> resultList = new ArrayList<ScriptedURL>();
 
     Cursor cursor = dataBase.rawQuery("SELECT * from page ", null);
     while (cursor.moveToNext()) {
-      ScriptedURL scriptecItem = new ScriptedURL(cursor.getString(2), cursor.getInt(1));
-      resultList.add(scriptecItem);
+      ScriptedURL scriptedItem = new ScriptedURL(cursor.getString(2), cursor.getInt(1), cursor.getInt(3));
```
Comment (xhae) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: Can you write some explanation about those indexes?

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java [hunk 3]
```diff
@@ -47,14 +47,27 @@ public int getReadPageCount() {
...
+    return resultList;
+  }
+
+  public ArrayList<ScriptedURL> getUnreadUrlList() {
+    SQLiteDatabase dataBase = getReadableDatabase();
+    ArrayList<ScriptedURL> resultList = new ArrayList<ScriptedURL>();
+
+    Cursor cursor = dataBase.rawQuery("SELECT * from page where read = 0", null);
+    while (cursor.moveToNext()) {
+      ScriptedURL scriptedItem = new ScriptedURL(cursor.getString(2), cursor.getInt(1), cursor.getInt(3));
```
Comment (xhae) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: ditto

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java (hunk 2 above)
Comment (sunju0123) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: we curosr 3 means word count value.
Soyee request word count value.

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java (hunk 3 above)
Comment (xhae) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: It looks like `getAllUrlList` and `getUnreadUrlList` shares most of the logic. Can we unify those for the better readability?

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java (hunk 1 above)
Comment (sunju0123) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: in google code style, don't make empty block 
https://google.github.io/styleguide/javaguide.html#s4.1.2-braces-empty-blocks

I want to checking url duplication by try-catch.. 
how to change this line looks good? Add failed log? or I think get all url list and check duplication.

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java (hunk 2 above)
Comment (sunju0123) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: and I found a typing error **scriptecItem**

so change 
scriptecItem -> scriptedItem

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java (hunk 3 above)
Comment (sunju0123) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: how about this code?
adding getUrlListFromQuery method.

```
  public ArrayList<ScriptedURL> getAllUrlList() {
    String unreadQuery = "SELECT * from page";
    return getUrlListFromQuery(unreadQuery);
  }

  public ArrayList<ScriptedURL> getUnreadUrlList() {
    String unreadQuery = "SELECT * from page where read = 0";
    return getUrlListFromQuery(unreadQuery);
  }

  private ArrayList<ScriptedURL> getUrlListFromQuery(String query) {
    SQLiteDatabase dataBase = getReadableDatabase();
    ArrayList<ScriptedURL> resultList = new ArrayList<ScriptedURL>();

    Cursor cursor = dataBase.rawQuery(query, null);
    while (cursor.moveToNext()) {
      ScriptedURL scriptedItem = new ScriptedURL(cursor.getString(2), cursor.getInt(1), cursor.getInt(3));
      resultList.add(scriptedItem);
    }

    return resultList;
  }
```

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java (hunk 1 above)
Comment (xhae) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: Yup. The problem in here is that the code is not handling the exception so we have a empty block. Handle it in any way so we don't skip the exception.

Comment (xhae) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: BTW I didn't know about the nonempty block rules neither. Thanks for the information 👍

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java [hunk 4]
```diff
@@ -86,11 +86,16 @@ public void setWordCount(String url, int wordCount) {
...
-    try {
-      dataBase.execSQL("INSERT INTO PAGE (URL) VALUES ('" + url + "');");
-      dataBase.close();
-    } catch (Exception exception) {
-      dataBase.close();
+    dataBase.execSQL("INSERT INTO PAGE (URL) VALUES ('" + url + "');");
+    dataBase.close();
+  }
+
+  public void insertScriptedDataWithCheckDuplication(String url) {
```
Comment (sunju0123) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: I create temporarily method `insertScriptedDataWithCheckDuplication`
please using this method when insert url with no duplication.

but I think 

> scan new feedly url list -> check duplication url with database url list and insert.

is more good performance 
@leesera

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java (hunk 1 above)
Comment (sunju0123) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: I add new method. It is checking url in every url insert.
and set back insertScriptedData method (no try catch)

File: Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java (hunk 2 above)
Comment (xhae) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: Please leave those as a comment in the source or using some const integer would be also fine.

Comment (sunju0123) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: I got it!

Comment (sunju0123) [inline Roys/app/src/main/java/com/dwg_karrier/roys/DataBaseOpenHelper.java]: I add const number and change that line in 5th commit.

**Comment by xhae** [comment]
Thank you for the contributions! 👍

**Pull request merged**